### ISSUE TITLE
delete row: Could not determine name of table

### ISSUE THREAD

**Issue by EugenNY** [HeidiSQL/HeidiSQL#860]
## Steps to reproduce this issue

i'm using a mysql instance via "network type: mariadb or mysql" and running a query like this:
```
SELECT * FROM (
    SELECT id,in_id, upper(concat_ws(",",
        if(the_flags!="", the_flags, NULL),
        if(flag1="true", "flag1", NULL),
        if(flag2="true", "flag2", NULL),
        if(flag10="true", "flag10", NULL),
    )) as the_flags, url, DATE
    from the_table
    where has_flag = 'true'
    and id > 40072
) AS x
WHERE  the_flags = "FLAG1,FLAG2"
order by id DESC LIMIT 200;
```

now I want to delete a single or multiple rows in the results in the grid view, but it fails.

## Current behavior

this query used to work for 2 years with HeidSQL, but recently I updated to a newer version and now I get this error message when I try to delete any row from that query's result grid view:
![image](https://user-images.githubusercontent.com/7329265/72719218-b5796400-3b77-11ea-8c9b-880908db6dce.png)

## Expected behavior

It should just delete the row or multiple rows. In this case, it would delete rows from the table `the_table`

## Environment

Windows 10, MySQL

* **HeidiSQL version:** 
Version 10.3.0.5837 (64 Bit)

* **Operating system:** 
Windows 10

**Comment by ansgarbecker** [comment]
I made bigger changes to the code for column and key retrieval, mainly for a code cleanup and for being able to add SQLite support. However, it may now throw such errors, although the MySQL logic has not changed so much.
Could you please post the `CREATE TABLE` code and one or two `INSERTs`?

**Comment by EugenNY** [comment]
i need to make a pseudo-table, to not leak any sensitive data, so i hope i didnt make any mistakes:
```
CREATE TABLE `the_table` (
	`id` INT(11) NOT NULL AUTO_INCREMENT,
	`user` INT(11) NULL DEFAULT NULL,
	`in_id` BIGINT(20) NOT NULL DEFAULT NULL,
	`flag1` ENUM('true','false') NULL DEFAULT NULL COLLATE 'utf8_general_ci',
	`flag2` ENUM('true','false') NULL DEFAULT NULL COLLATE 'utf8_general_ci',
	`flag10` ENUM('true','false') NULL DEFAULT NULL COLLATE 'utf8_general_ci',
	`url` MEDIUMTEXT NULL DEFAULT NULL COLLATE 'utf8_general_ci',
	`date` DATETIME NULL DEFAULT 'CURRENT_TIMESTAMP',
	PRIMARY KEY (`id`) USING BTREE,
	INDEX `in_id` (`in_id`) USING BTREE,
	INDEX `the_table_user` (`user`) USING BTREE
)
COLLATE='utf8_general_ci'
ENGINE=InnoDB
AUTO_INCREMENT=519738
;
```

```INSERT INTO the_table (`user`) VALUES ('100');```
```INSERT INTO the_table (`user`) VALUES ('200');```

let me know if you can reproduce it with that.

**Comment by ansgarbecker** [comment]
Yes, I can reproduce it, with some modifications in your code. I also get that "_could not determine name of table_" error.

Renaming the subselect alias from "x" to "the_table" solved that for me.

However, I wonder why that did not throw errors in previous HeidiSQL versions. Perhaps it's also a matter of the server version, and/or the client library (libmysql.dll or libmariadb.dll). You could try to set a different one in your session setting:
![grafik](https://user-images.githubusercontent.com/7986591/72739402-dc4d8f80-3ba3-11ea-9fd4-ab9b95e6b8db.png)

**Comment by EugenNY** [comment]
thanks for the quick feedback. very helpful!

i tried libmysql (both), but it's still the same...

good news however: indeed renaming the subselect alias from "x" to "the_table" solved it for me too!

**Comment by ansgarbecker** [comment]
Ok, I think the remaining issue would be solved within #839, so I'll close this one.
